community/server/src/main/java/org/neo4j/server/rest/web/InternalJettyServletRequest.java

```diff
@@ -361,4 +358,50 @@ public HttpChannelState getHttpChannelState()
     {
         return DUMMY_HTTP_CHANNEL_STATE;
     }
+
+    public static class RequestData
+    {
+        public final String remoteAddress;
+        public final String remoteHost;
+        public final boolean isSecure;
+        public final int remotePort;
+        public final String localName;
+        public final String localAddress;
+        public final int localPort;
+        public final String authType;
+
+        public RequestData(
+                String remoteAddress,
+                String remoteHost,
+                boolean isSecure,
+                int remotePort,
+                String localName,
+                String localAddress,
+                int localPort,
+                String authType
+        ) {
+            this.remoteAddress = remoteAddress;
+            this.remoteHost = remoteHost;
+            this.isSecure = isSecure;
+            this.remotePort = remotePort;
+            this.localName = localName;
+            this.localAddress = localAddress;
+            this.localPort = localPort;
+            this.authType = authType;
+        }
+
+        public static RequestData from( HttpServletRequest req )
+        {
+            return new RequestData(
+                    req.getRemoteAddr(),
+                    req.getRemoteHost(),
+                    req.isSecure(),
+                    req.getRemotePort(),
+                    req.getLocalName(),
+                    req.getLocalAddr(),
+                    req.getLocalPort(),
+                    req.getAuthType()
```

Comment:
Yes, but this seems not to be even used. Possibly there's a need for some large clean-ups here, but I don't think we should perform that as a part of this PR. Essentially any null-Strings gotten from the Request will simply be passed on transparently to InternalJettyServletRequest users.